### ISSUE TITLE
import-url from bucket with folder names that are prefixes of another folder doesn't work as expected

### ISSUE THREAD

**Issue by D34D10CK** [iterative/dvc-gs#23]
# Bug Report

## Description

When using the command `import-url` to import a folder from a Google cloud storage bucket that is a substring of another folder, it also import that other folder.
Example:
consider the following bucket structure:
```
my-bucket/data/file1.txt
my-bucket/dataset/file2.txt
```
Then the command `dvc import-url "gs://my-bucket/data" .` will also import my-bucket/dataset.

### Reproduce

1. Create a bucket `my-bucket` with two folders `data` and `dataset`
2. In a dvc project, run `dvc import-url "gs://my-bucket/data" .` 

### Expected

Only `my-bucket/data` should be imported.

### Environment information

**Output of `dvc doctor`:**

```console
$ dvc doctor
```
DVC version: 2.35.2 (pip)
---------------------------------
Platform: Python 3.10.8 on Linux-4.19.0-22-cloud-amd64-x86_64-with-glibc2.28
Subprojects:
        dvc_data = 0.28.0
        dvc_objects = 0.14.0
        dvc_render = 0.0.14
        dvc_task = 0.1.6
        dvclive = 1.0.1
        scmrepo = 0.1.3
Supports:
        gs (gcsfs = 2022.11.0),
        http (aiohttp = 3.8.3, aiohttp-retry = 2.8.3),
        https (aiohttp = 3.8.3, aiohttp-retry = 2.8.3)
Cache types: hardlink, symlink
Cache directory: ext4 on /dev/sda1
Caches: local
Remotes: gs
Workspace directory: ext4 on /dev/sda1
Repo: dvc, git

**Additional Information (if any):**

**Comment by pmrowla** [comment]
@D34D10CK can you try the latest release? There have been some changes in both dvc and the underlying gcsfs library that may have resoled your issue

**Comment by pmrowla** [comment]
Can reproduce in latest main + dvc-gs with and without the trailing `/`. Only the expected folder is imported (the resulting `.dvc` file only contains the single intended directory). The issue is that additional objects which match the given prefix are downloaded into the workspace (but they are not included in the resulting `.dvc` file).

**Comment by pmrowla** [comment]
This is fixed in `dvc-gs==2.22.1`
